lib/Tumblr/StreamBuilder/StreamFilters/StreamFilter.php

```diff
@@ -123,6 +123,9 @@ final public function filter(
             foreach ($filtered->get_released() as $element) {
                 $tracer->release_element($this, $element);
             }
+            if (count($elements) === $filtered->get_released_count()) {
+                $tracer->release_all_elements($this, $filtered->get_released_count());
+            }
```

Comment:
rather than this can we report the % of released elements ?

Reply:
if we report the % of released elements, do we need the release_all tick?

this would be the diff for the % of released elements implementation:

```
diff --git a/lib/Tumblr/StreamBuilder/StreamFilters/StreamFilter.php b/lib/Tumblr/StreamBuilder/StreamFilters/StreamFilter.php
index 621cb17..70a772e 100644
--- a/lib/Tumblr/StreamBuilder/StreamFilters/StreamFilter.php
+++ b/lib/Tumblr/StreamBuilder/StreamFilters/StreamFilter.php
@@ -118,7 +118,8 @@ abstract class StreamFilter extends Templatable
             $tracer->end_filter(
                 $this,
                 $filtered->get_released_count(),
-                [$t0, microtime(true) - $t0]
+                [$t0, microtime(true) - $t0],
+                count($elements)
             );
             foreach ($filtered->get_released() as $element) {
                 $tracer->release_element($this, $element);
diff --git a/lib/Tumblr/StreamBuilder/StreamTracers/StreamTracer.php b/lib/Tumblr/StreamBuilder/StreamTracers/StreamTracer.php
index 8963fc2..8736ad3 100644
--- a/lib/Tumblr/StreamBuilder/StreamTracers/StreamTracer.php
+++ b/lib/Tumblr/StreamBuilder/StreamTracers/StreamTracer.php
@@ -387,21 +387,28 @@ abstract class StreamTracer
      * @param StreamFilter $filter The stream filter is applied.
      * @param int $released_count The number of items being filtered.
      * @param array $timing Zero indexed tuple of the start time and duration of the operation (in that order)
+     * @param int|null $total_count The number of elements that went through the filter step, out of which $release_count were filtered.
      * @return void
      */
     final public function end_filter(
         StreamFilter $filter,
         int $released_count,
-        array $timing
+        array $timing,
+        ?int $total_count = 0
     ): void {
+        $meta = [static::META_COUNT => $released_count];
+
+        if ($total_count > 0) {
+            $ratio = $released_count / $total_count;
+            $meta['released_ratio'] = $ratio;
+        }
+
         $this->trace_event(
             static::CATEGORY_FILTER,
             $filter,
             static::EVENT_END,
             $timing,
-            [
-                static::META_COUNT => $released_count,
-            ]
+            $meta
         );
     }
```

Reply:
> if we report the % of released elements, do we need the release_all tick?

I would keep the release_all tick because for our use-case it's more easy to monitor that, but % release is more informative.